syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";

package serialization;

message Coordinates {
	double        lat = 1;
    double        lng = 2;
}

message Stop {			
	string name = 1;
	Coordinates coordinates = 2;
	repeated uint32 buses_id = 3;
}

message Bus {
	string name = 1;
	repeated uint32 stops_id = 2;
	bool round_trip = 3;	
	uint32 end_stop_id = 4;
}

message DistanceBetweenStops {
	uint32 from = 1;
	uint32 to = 2;
    uint32 distance = 3;    
}

message CatalogueData {
    repeated Bus buses_base = 1;
    repeated Stop stops_base = 2;
	repeated DistanceBetweenStops dist_btw_stops = 3;   
}

message TransportCatalogue {
	CatalogueData catalogue = 1;
	MapRenderer map_renderer = 2;
	TransportRouter transport_router = 3;
}
